docker/ubuntu/bionic/user-config.jam

```diff
@@ -0,0 +1,22 @@
+#
+# Copyright (C) 2018 James E. King III
+#
+# Use, modification, and distribution are subject to the
+# Boost Software License, Version 1.0. (See accompanying file
+# LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+#
+
+# Help bjam find clang++-6.0 for "clang"
+# (see: https://github.com/boostorg/build/issues/306)
+using clang : 6.0 : /usr/bin/clang++-6.0 ;
+using gcc : 7.3 : /usr/bin/g++-7 ;
+
```

Comment:
do we want to add more compiler configurations for users to work with, such as

```
using gcc : 7~c++17 : /usr/bin/g++-7 : <cxxflags>"-std=c++17" ;
using gcc : 7~gnu17 : /usr/bin/g++-7 : <cxxflags>"-std=gnu++17" ;
using gcc : 7~c++2a : /usr/bin/g++-7 : <cxxflags>"-std=c++2a" ;
using gcc : 7~gnu2a : /usr/bin/g++-7 : <cxxflags>"-std=gnu++2a" ;
using gcc : 7~c++2a~O2 : /usr/bin/g++-7 : <cxxflags>"-std=c++2a -O2" ;
using gcc : 7~c++2a~warn : /usr/bin/g++-7 : <cxxflags>"-std=c++2a -Wall -Wextra" ;

```

Reply:
I used to do this, but I believe the new hotness is to specify it on the command line:

    ./b2 --toolset=gcc/cxxstd=17

Reply:
I added the user-config.jam because without it I couldn't say "<toolset>gcc" or "<toolset>clang" and have it work from within the docker instance.  For clang this is because the clang selector logic in bjam is pretty old and hasn't been updated with the gcc version selector "hotness".  I'd prefer not to have a user-profile.jam at all inside the docker instance and instead use toolset=clang-6 cxxstd=14 cxxstd-dialect=gnu, if you want to use /usr/bin/clang++-6 with -std=gnu++14.

In fact I think it might be ideal in the docker environment to take the user-config.jam from the user's home directory and put it into the docker instance at startup time... that way you can manage your own settings the way you are used to managing them, without having to copy things around yourself after you spin up the docker instance.